### PR TITLE
Add PrepareParams into ProcessInfo

### PR DESCRIPTION
**Pull request by zengyan1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #17200 

## What this PR does / why we need it:
Add PrepareParams into ProcessInfo

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Added `prepareParams` field to `processHelper` struct and handled its initialization and usage in `remoterunServer.go`.
- Updated `BuildProcessInfo` and `Decode` functions in `process_codec.go` to handle `prepareParams`.
- Simplified struct initialization in `mysql_sql.go` parser code.
- Added `prepare_params` field to `ProcessInfo` message in `pipeline.proto`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>remoterunServer.go</strong><dd><code>Add and handle `prepareParams` in processHelper struct</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/compile/remoterunServer.go

<li>Added <code>prepareParams</code> field to <code>processHelper</code> struct.<br> <li> Set <code>prepareParams</code> in <code>newCompile</code> function.<br> <li> Unmarshal <code>prepareParams</code> in <code>generateProcessHelper</code> function.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17332/files#diff-d73b7f89321871c12e2595b1b3fd175f2f3a8a3cff9e153f69c02f23cd650d6f">+12/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>process_codec.go</strong><dd><code>Handle `prepareParams` in process codec functions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/vm/process/process_codec.go

<li>Marshal <code>prepareParams</code> in <code>BuildProcessInfo</code> function.<br> <li> Unmarshal <code>prepareParams</code> in <code>Decode</code> function.<br>

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17332/files#diff-4deb7a07a8e6de95524fd7ec7596f7bbd8aec2fab3d10d2fb57458ee0c490fec">+15/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pipeline.proto</strong><dd><code>Add `prepare_params` field to ProcessInfo message</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

proto/pipeline.proto

- Added `prepare_params` field to `ProcessInfo` message.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17332/files#diff-e9c91df4d87ffef63693440b86da7db2a8349c550c6c789bcc8e71f3e81309a6">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>mysql_sql.go</strong><dd><code>Simplify struct initialization in MySQL parser</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/sql/parsers/dialect/mysql/mysql_sql.go

- Simplified struct initialization in parser code.

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17332/files#diff-6aa8b3d2cec1bd35bafe74d27392cee02153f7b4ec574d40eff17db7c30d41e2">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Additional files (token-limit)</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pipeline.pb.go</strong><dd><code>...</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/pb/pipeline/pipeline.pb.go

...

</details>

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/17332/files#diff-966f3a64eaa6e8c3a8815332fec622b1e79c9d999ed32198a74e0038c17a25ec">+348/-294</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zengyan1** [comment]
Testing

**Comment by zengyan1** [comment]
已添加bvt case，可在
分支：`main` commit：`78e277a67a33d3a425e0ae0066cc035b6d862743`
多cn环境稳定panic

**Review by heni02 — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Review by m-schen — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

File: pkg/sql/plan/shuffle.go [hunk 1]
```diff
@@ -339,23 +339,28 @@ func determinShuffleForJoin(n *plan.Node, builder *QueryBuilder) {
 	}
 
 	// get the column of left child
-	var expr *plan.Expr
+	var expr0, expr1 *plan.Expr
 	cond := n.OnList[idx]
 	switch condImpl := cond.Expr.(type) {
 	case *plan.Expr_F:
-		expr = condImpl.F.Args[0]
```
Comment (fengttt) [inline pkg/sql/plan/shuffle.go:346]: What happens to function with 3 args?

**Review by fengttt — CHANGES_REQUESTED** [review]

File: pkg/sql/plan/shuffle.go (hunk 1 above)
Comment (zengyan1) [reply pkg/sql/plan/shuffle.go:346]: > What happens to function with 3 args?

The join condition will never have 3 args, it's like "left.col = right.col" or "Function(left.col) < Function(right.col)"

**Review by aunjgr — APPROVED** [review]

**Review by zhangxu19830126 — APPROVED** [review]

**Comment by zengyan1** [comment]
Testing

**Review by fengttt — APPROVED** [review]

**Pull request merged**